### PR TITLE
Fixes Brave Ads landed events should only be triggered if the user clicked the ad

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/12155

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [x] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [ ] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

See https://github.com/brave/brave-browser/issues/12155

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Review by emerick — APPROVED** [review]
LGTM

File: vendor/bat-native-ads/src/bat/ads/internal/ad_events/ad_notification_event_clicked.cc [hunk 1]
```diff
@@ -34,6 +34,8 @@ void AdNotificationEventClicked::Trigger(
   ads_->get_ad_notifications()->Remove(ad_notification.uuid,
       /* should dismiss */ true);
 
+  ads_->set_last_shown_ad_notification(ad_notification);
```
Comment (moritzhaller) [inline vendor/bat-native-ads/src/bat/ads/internal/ad_events/ad_notification_event_clicked.cc:37]: nit: If I understand correctly the corresponding variable in `AdsImpl` is only set on click now. So maybe rename to `set_last_clicked_ad_notification()`  and corresponding to that in `AdsImpl` rename `last_shown_ad_notification_` to `last_clicked_ad_notification_`.

**Review by moritzhaller — APPROVED** [review]
lgtm!

File: vendor/bat-native-ads/src/bat/ads/internal/ad_events/ad_notification_event_clicked.cc (hunk 1 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/ad_events/ad_notification_event_clicked.cc:37]: Renaming to `last_shown_ad_` in an upcoming PR to support different ad types

**Review by yachtcaptain23 — APPROVED** [review]

**Pull request merged**